ibis/backends/sql/compilers/flink.py

```diff
@@ -575,5 +575,21 @@ def visit_MapMerge(self, op: ops.MapMerge, *, left, right):
     def visit_StructColumn(self, op, *, names, values):
         return self.cast(sge.Struct(expressions=list(values)), op.dtype)
 
+    def visit_ArrayCollect(self, op, *, arg, where, order_by, include_null):
+        if order_by:
+            raise com.UnsupportedOperationError(
+                "ordering of order-sensitive aggregations via `order_by` is "
+                "not supported for this backend"
+            )
+        # the only way to get filtering *and* respecting nulls is to use
+        # `FILTER` syntax, but it's broken in various ways for other aggregates
+        out = self.f.array_agg(arg)
+        if not include_null:
```

Comment:
Note that Flink does support `IGNORE NULLS` for this operation, but it's broken if the argument is anything more complicated than a direct column reference. Using manual filtering like you're doing here is the best we can do for now.